src/typescript/azure-openapi-validator/index.ts

```diff
@@ -36,48 +39,64 @@ export function run(
       openapiDefinition,
       rule.appliesTo_JsonQuery || "$"
     )) {
-      for (const message of rule.run(
-        openapiDefinition,
-        section.value,
-        section.path.slice(1)
-      )) {
-        const readableCategory = rule.category;
+      if (rule.run) {
+        for (const message of rule.run(
+          openapiDefinition,
+          section.value,
+          section.path.slice(1))) {
+          handle(rule, message)
+        }
+      }
+      else {
+        for await (const message of rule.asyncRun(
```

Comment:
Why do we need aysnrun here?

Reply:
so that the  await  can be used in the rule function